edx_sga/tasks.py

```diff
@@ -0,0 +1,309 @@
+from uuid import uuid4
+import six
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.files.storage import default_storage  # lint-amnesty, pylint: disable=import-error
+
+from lms import CELERY_APP  # pylint: disable=no-name-in-module, import-error
+from submissions import api as submissions_api  # lint-amnesty, pylint: disable=import-error
+from submissions.models import StudentItem as SubmissionsStudent  # lint-amnesty, pylint: disable=import-error
+from student.models import user_by_anonymous_id  # lint-amnesty, pylint: disable=import-error
+
+log = logging.getLogger(__name__)
+DATA_DIR = getattr(settings, "DATA_DIR", "/edx/app/edxapp/data")
+
+
+def start_downloading_submissions(course_id, block_id, location, user):
+    """
+    helper to start downloading task
+
+    Args:
+        course_id (unicode): edx course id
+        block_id (unicode): edx block id
+        location (Location): location of sga module
+        user (django.contrib.auth.models.Use): staff user
+    """
+    if course_id and block_id and location:
+        zip_student_submissions.delay(
+            course_id,
+            block_id,
+            location,
+            user.username
+        )
+
+
+def _mk_dir(path):
+    """
+    Create a new directory.
+
+    Args:
+        path (str): path of new directory
+    """
+    _rm_file(path)
+    os.mkdir(path)
+
+
+def _rm_file(path):
+    """
+    Removes file if it exist.
+
+    Args:
+        path (str): path of new directory
+    """
+    if os.path.exists(path):
+        rmtree(path)
+
+
+def get_user_by_id(user_id):
+    """
+    Returns user object
+
+    Args:
+        user_id (int): user id
+    """
+    try:
+        user = User.objects.get(id=user_id)
+    except User.DoesNotExist:
+        user = None
+
+    return user
+
+
+def _get_user_name(user):
+    """
+    returns user name
+
+    Args:
+        user (django.contrib.auth.models.User): user instance
+
+    return:
+        (str): user name
+    """
+    if user:
+        user_name = user.username
+    else:
+        user_name = uuid4().hex
+
+    return user_name
+
+
+def user_name_by_id(user_id):
+    """
+    returns user name by given user id
+
+    Args:
+        user_id (int): user id
+
+    return:
+        (str): user name
+    """
+    if user_id:
+        user = get_user_by_id(user_id)
+        user_name = _get_user_name(user)
+    else:
+        user_name = uuid4().hex
+
+    return user_name
+
+
+def _get_submissions(student_id, course_id, block_id):
+    """
+    Returns submissions of student
+
+    Args:
+        course_id (unicode): edx course id
+        block_id (unicode): edx block id
+    """
+    params = {
+        "student_id": student_id,
+        "course_id": course_id,
+        "item_id": block_id,
+        "item_type": 'sga',
+    }
+    return submissions_api.get_submissions(params)
+
+
+def _compress_student_submissions(block_id, course_id, location, destination_path):
```

Comment:
It looks like this function isn't actually doing any compressing. It's just copying files to a new folder to prepare for the compression, right? Maybe a better name would be `_collect_student_submissions`